app/code/Magento/Cron/Cron/CleanOldJobs.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Cron\Cron;
+
+use Magento\Config\App\Config\Type\System;
+use Magento\Cron\Model\DeadlockRetrierInterface;
+use Magento\Cron\Model\ResourceModel\Schedule;
+use Magento\Cron\Model\ScheduleFactory;
+use Magento\Cron\Observer\ProcessCronQueueObserver;
+use Magento\Framework\App\Config;
+use Magento\Framework\Stdlib\DateTime\DateTime;
+
+class CleanOldJobs
+{
+    /**
+     * This docblock provides no new information
+     *
+     * @param Config $config
+     * @param DateTime $dateTime
+     * @param DeadlockRetrierInterface $retrier
+     * @param ScheduleFactory $scheduleFactory
+     */
+    public function __construct(
+        private readonly Config $config,
+        private readonly DateTime $dateTime,
+        private readonly DeadlockRetrierInterface $retrier,
+        private readonly ScheduleFactory $scheduleFactory,
+    ) {
+    }
+
+    /**
+     * Run the 'clean_cron_schedule' cronjob
+     *
+     * @return void
+     */
+    public function execute(): void
```

Comment:
Can you clarify why we need the logic on top of https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Cron/Observer/ProcessCronQueueObserver.php#L562

Reply:
@kandy the linked code only cleans up known jobs (see line 596) in known groups (see line 558)

https://github.com/magento/magento2/blob/995a05c9c7bf661729d79a01ad1447370bf4f39f/app/code/Magento/Cron/Observer/ProcessCronQueueObserver.php#L596
https://github.com/magento/magento2/blob/995a05c9c7bf661729d79a01ad1447370bf4f39f/app/code/Magento/Cron/Observer/ProcessCronQueueObserver.php#L558

Reply:
ok, make sense 
thanks you for clarification